client/my-sites/comments/controller.js

```diff
@@ -9,10 +9,17 @@ import { includes } from 'lodash';
 /**
  * Internal dependencies
  */
-import route from 'lib/route';
 import CommentsManagement from './main';
+import config from 'config';
+import route from 'lib/route';
 
-const VALID_STATUSES = [ 'pending', 'approved', 'spam', 'trash', 'all' ];
+const VALID_STATUSES = [
+	'pending',
+	'approved',
+	'spam',
+	'trash',
+	...config.isEnabled( 'comments/management/all-list' ) && 'all',
```

Comment:
Hmm, not sure what behavior I would expect from a destructure on a boolean. As is if the flag is not enabled, do we have an extra boolean in VALID_STATUSES?